google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/SchemaCompact.java

```diff
@@ -38,8 +46,32 @@
   private static SchemaCompact compact;
   private static String tablePatternString = "projects/([^/]+)/datasets/([^/]+)/tables/([^/]+)";
   private static Pattern tablePattern = Pattern.compile(tablePatternString);
+  private static final int NestingLimit = 15;
+  // private static Set<Descriptors.FieldDescriptor.Type> SupportedTypesHashSet =
+
+  private static Set SupportedTypes =
+      Collections.unmodifiableSet(
+          new HashSet<>(
+              Arrays.asList(
+                  Descriptors.FieldDescriptor.Type.INT32,
+                  Descriptors.FieldDescriptor.Type.INT64,
+                  Descriptors.FieldDescriptor.Type.UINT32,
+                  Descriptors.FieldDescriptor.Type.UINT64,
+                  Descriptors.FieldDescriptor.Type.FIXED32,
+                  Descriptors.FieldDescriptor.Type.FIXED64,
+                  Descriptors.FieldDescriptor.Type.SFIXED32,
+                  Descriptors.FieldDescriptor.Type.SFIXED64,
+                  Descriptors.FieldDescriptor.Type.FLOAT,
+                  Descriptors.FieldDescriptor.Type.DOUBLE,
+                  Descriptors.FieldDescriptor.Type.BOOL,
+                  Descriptors.FieldDescriptor.Type.BYTES,
+                  Descriptors.FieldDescriptor.Type.STRING,
+                  Descriptors.FieldDescriptor.Type.MESSAGE,
+                  Descriptors.FieldDescriptor.Type.GROUP,
+                  Descriptors.FieldDescriptor.Type.ENUM)));
 
   private SchemaCompact(BigQuery bigquery) {
+    // TODO: Add functionality that allows SchemaCompact to build schemas.
```

Comment:
is there a reason for us to keep this TODO here?

Reply:
Yup, Yiru asked me to leave that there as that should be a functionality we should add in, but not related to the current PR.

Reply:
ok sg